utility/src/main/java/org/oppia/util/parser/UrlImageParser.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.util.parser
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.text.Html
+import android.widget.TextView
+import com.bumptech.glide.request.target.SimpleTarget
+import com.bumptech.glide.request.transition.Transition
+import dagger.Binds
+import org.oppia.util.R
+import javax.inject.Inject
+import javax.inject.Singleton
+
+// TODO(#169): Replace this with exploration asset downloader.
+/** UrlImage Parser for android TextView to load Html Image tag. */
+class UrlImageParser private constructor(
+  @ApplicationContext private val context: Context,
+  @DefaultGcsPrefix private val gcsPrefix: String,
+  @DefaultGcsResource private val gcsResource: String,
+  @ImageDownloadUrlTemplate private var imageDownloadUrlTemplate: String,
+  private val htmlContentTextView: TextView,
+  private val entityType: String,
+  private val entityId: String,
+  @ImageLoaderAnnotation private val imageLoader:ImageLoader
+) : Html.ImageGetter {
+
+
+  /***
+   * This method is called when the HTML parser encounters an <img> tag.
+   * @param urlString : urlString argument is the string from the "src" attribute.
+   * @return Drawable : Drawable representation of the image.
+   */
+  override fun getDrawable(urlString: String): Drawable {
+    imageDownloadUrlTemplate = String.format(imageDownloadUrlTemplate, entityType, entityId, urlString)
+    val urlDrawable = UrlDrawable()
+    val target = BitmapTarget(urlDrawable)
+    imageLoader.load(
+      context,
+      gcsPrefix + gcsResource + imageDownloadUrlTemplate,
+      target
+    )
+    return urlDrawable
+  }
+
+  private inner class BitmapTarget(private val urlDrawable: UrlDrawable) : SimpleTarget<Bitmap>() {
+    internal var drawable: Drawable? = null
+
+    override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
+      drawable = BitmapDrawable(context.resources, resource)
```

Comment:
Let's make this a local variable. I don't see a reason why it needs to be a class variable.

Reply:
Done.